### DISCUSSION TITLE
Let enum be iterable

### DISCUSSION THREAD

**Discussion by smartiniOnGitHub** [vlang/v#12472] [Ideas]
Hi all,
what do you think to be able to iterate across all enum values ? Maybe generated at compile time only when needed.

It would be useful to loop in a simple way with bounded values etc ...
If possible maybe with the ability to know current item index (like in other for loops) and even: numeric value (from enum value), string value (from enum string key).

Or if simpler, a function (a compile time facility ?) to return an array with those values, and use it as iterable in for loops etc ...

I think it could be related/an expansion of what proposed in:
https://github.com/vlang/v/discussions/11646

Hope you like this proposal.

Thanks. Regards

**Comment by programandala-net** [comment]
I just missed that feature, converting a program of mine from Odin to V.

I've searched the docs but it seems there's no way to iterate an `enum` in V yet.  I will rewrite my code with arrays or something.

**Reply by felipensp** [reply]
Iterator over enum values is already possible, see https://github.com/vlang/v/blob/master/doc/docs.md#values

**Reply by programandala-net** [reply]
Thank you. I had searched https://docs.vlang.io/type-declarations.html#enums and found nothing.
